desktop-widgets/divelogexportdialog.cpp

```diff
@@ -275,31 +274,77 @@ void DiveLogExportDialog::export_TeX(const char *filename, const bool selected_o
 
 		QString star = "*";
 		QString viz = star.repeated(dive->visibility);
-		int i;
+		QString rating = star.repeated(dive->rating);
 
-		for (i = 0; i < MAX_CYLINDERS; i++)
-			if (is_cylinder_used(dive, i))
-				delta_p.mbar += dive->cylinder[i].start.mbar - dive->cylinder[i].end.mbar;
+		int i;
+		int qty_cyl;
+		int qty_weight;
+		float total_weight;
 
 		if (need_pagebreak)
 			put_format(&buf, "\\vfill\\eject\n");
 		need_pagebreak = true;
 		put_format(&buf, "\\def\\date{%04u-%02u-%02u}\n",
 		      tm.tm_year, tm.tm_mon+1, tm.tm_mday);
 		put_format(&buf, "\\def\\number{%d}\n", dive->number);
-		put_format(&buf, "\\def\\place{%s}\n", site ? site->name : "");
-		put_format(&buf, "\\def\\spot{}\n");
+		put_format(&buf, "\\def\\sitename{%s}\n", site ? site->name : "");
+		put_format(&buf, "\\def\\gpslat{%f}\n", site ? site->latitude.udeg / 1000000.0 : 0);
+		put_format(&buf, "\\def\\gpslon{%f}\n", site ? site->longitude.udeg / 1000000.0 : 0);
+		put_format(&buf, "\\def\\computer{%s}\n", dive->dc.model);
 		put_format(&buf, "\\def\\country{%s}\n", country.toUtf8().data());
-		put_format(&buf, "\\def\\entrance{}\n");
 		put_format(&buf, "\\def\\time{%u:%02u}\n", FRACTION(dive->duration.seconds, 60));
-		put_format(&buf, "\\def\\depth{%u.%01um}\n", FRACTION(dive->maxdepth.mm / 100, 10));
-		put_format(&buf, "\\def\\gasuse{%u.%01ubar}\n", FRACTION(delta_p.mbar / 100, 10));
+		put_format(&buf, "\\def\\maxtemp{%.2f}\n", (dive->maxtemp.mkelvin) ? (dive->maxtemp.mkelvin / 1000.0) - 273.15 : 0);
+		put_format(&buf, "\\def\\mintemp{%.2f}\n", (dive->mintemp.mkelvin) ? (dive->mintemp.mkelvin / 1000.0) - 273.15 : 0);
+		put_format(&buf, "\\def\\watertemp{%.2f}\n", (dive->watertemp.mkelvin) ? (dive->watertemp.mkelvin / 1000.0) - 273.15 : 0);
+		put_format(&buf, "\\def\\airtemp{%.2f}\n", (dive->airtemp.mkelvin) ? (dive->airtemp.mkelvin / 1000.0) - 273.15 : 0);
+		put_format(&buf, "\\def\\maxdepth{%u.%01u m}\n", FRACTION(dive->maxdepth.mm / 100, 10));
+		put_format(&buf, "\\def\\avedepth{%u.%01u m}\n", FRACTION(dive->meandepth.mm / 100, 10));
+
+		//Code block prints start/end press. for all cylinders used, number of cyl. used, SAC, and total delta_p
+		qty_cyl = 0;
+		for (i = 0; i < MAX_CYLINDERS; i++)
+			if (is_cylinder_used(dive, i)){
+				put_format(&buf, "\\def\\cyl%ddescription{%s}\n", i + 1, dive->cylinder[i].type.description);
+				if (dive->cylinder[i].gasmix.o2.permille > 0){ //This code assumes that all gas must have oxygen, so a zero value indicated that the gas in question is AIR
```

Comment:
Have a look at what is in dive.c and dive.h. There are all kinds of handy helper functions. Here you probably want to use gassname(const struct *gasmix) defined in equipment.c If you really want to know the O2 fraction, use get_o2(), there is also get_he(). To print unitful quantities, you can use get_depth_units(int mm, int *frac, const char **units) and similar.